mmstudio/src/main/java/org/micromanager/data/internal/DefaultDataSaver.java

```diff
@@ -81,7 +81,10 @@ protected Void doInBackground() throws IOException {
       }
 
       final SummaryMetadata fSummary = summary;
-      duplicate_.setStorage(saver_);
+      if (!(saver_ instanceof NDTiffAdapter)) {
```

Comment:
Yes there's nothing for NDTiff to create until it has summary metadata. It expects it in the constructor and it is the first thing written to disk. I suppose this is somewhat different than how this API is designed, but if it works it works

Reply:
I dont really understand why there is a distinction between Datastore and Storage

Reply:
Datastore is common to all storages; Storage is different for RAM, single-page TIFF, multipage TIFF, etc. File (or memory) I/O is a separate concern from presenting a database-like container for images along axes. At least I'm guessing that was the idea; in reality DefaultDatastore knows way too much about the individual storages and the Storage interface is too close to Datastore so a lot of logic is duplicated anyway. And all of this makes too many assumptions about the storage format. Good to improve if anybody dares :)

Reply:
Ah got it, that makes sense